libminifi/src/core/repository/VolatileContentRepository.cpp

```diff
@@ -17,142 +16,133 @@
+  std::lock_guard lock(data_mtx_);
+  if (auto it = data_.find(claim.getContentFullPath()); it != data_.end()) {
+    return std::make_shared<StringRefStream>(it->second, data_mtx_, it->second, total_size_);
   }
-  logger_->log_info("Cannot write {} {}, returning nullptr to roll back session. Repo is either full or locked", claim.getContentFullPath(), size);
   return nullptr;
 }
 
 bool VolatileContentRepository::exists(const minifi::ResourceClaim &claim) {
-  std::lock_guard<std::mutex> lock(map_mutex_);
-  auto claim_check = master_list_.find(claim.getContentFullPath());
-  if (claim_check != master_list_.end()) {
-    auto ent = claim_check->second->takeOwnership();
-    return ent != nullptr;
-  }
-
-  return false;
+  std::lock_guard lock(data_mtx_);
+  return data_.contains(claim.getContentFullPath());
 }
 
-std::shared_ptr<io::BaseStream> VolatileContentRepository::read(const minifi::ResourceClaim &claim) {
-  std::lock_guard<std::mutex> lock(map_mutex_);
-  auto claim_check = master_list_.find(claim.getContentFullPath());
-  if (claim_check != master_list_.end()) {
-    auto ent = claim_check->second->takeOwnership();
-    if (ent == nullptr) {
-      return nullptr;
-    }
-    return std::make_shared<io::AtomicEntryStream<ResourceClaim::Path>>(claim.getContentFullPath(), ent);
-  }
+bool VolatileContentRepository::close(const minifi::ResourceClaim &claim) {
+  return remove(claim);
+}
 
-  return nullptr;
+void VolatileContentRepository::clearOrphans() {
+  // there are no persisted orphans to delete
 }
 
 bool VolatileContentRepository::removeKey(const std::string& content_path) {
-  if (LIKELY(minimize_locking_ == true)) {
-    std::lock_guard<std::mutex> lock(map_mutex_);
-    auto ent = master_list_.find(content_path);
-    if (ent != master_list_.end()) {
-      auto ptr = ent->second;
-      // if we cannot remove the entry we will let the owner's destructor
-      // decrement the reference count and free it
-      master_list_.erase(content_path);
-      // because of the test and set we need to decrement ownership
-      ptr->decrementOwnership();
-      if (ptr->freeValue(content_path)) {
-        logger_->log_info("Deleting resource {}", content_path);
-      } else {
-        logger_->log_info("free failed for {}", content_path);
-      }
-    } else {
-      logger_->log_info("Could not remove {}", content_path);
-    }
-  } else {
-    std::lock_guard<std::mutex> lock(map_mutex_);
-    auto claim_item = master_list_.find(content_path);
-    if (claim_item != master_list_.end()) {
-      auto size = claim_item->second->getLength();
-      delete claim_item->second;  // NOLINT(cppcoreguidelines-owning-memory)
-      master_list_.erase(content_path);
-      repo_data_.current_size -= size;
-    }
+  std::lock_guard lock(data_mtx_);
+  if (auto it = data_.find(content_path); it != data_.end()) {
+    total_size_ -= it->second->size();
+    data_.erase(it);
+    logger_->log_info("Deleting resource {}", content_path);
   }
```

Comment:
We could add some log if the key is not found

Reply:
added